### ISSUE TITLE
Cycle Profile: vehicular cycling

### ISSUE THREAD

**Issue by ferpg81** [osmandapp/OsmAnd#15393]
# 🚀 feature request

### Description

It's pretty simple: include another routing profile for bikes, that priorizes general vehicle lanes, roads, etc, over bike lanes/paths, etc. **A cycling profile for a vehicular cyclist**.

When I want to calculate a fast route to any destination by bike, the best for me is to select a car profile and then, see the result on screen carefully in order to identify, at a glance, roads with forbidden access to bikes, like motorways, and then, force recalculation with an intermediate _foo_ destination, avoiding these possible forbidden ways. It's not perfect at all, but instead, with cycle profile, the routing avoids roads like they were lava and always wants to bring you over any way that is dedicated specifically for bikes, tought these ways were like "shit" (it doesn't exist parameters that describe their usefullness on OSM, so all of them are treated equally on osmand, as useful and desirable paths for a cyclist). With default cycling profile, routing makes also walkarounds over simple avenues, and tries to avoid them selecting small and pretty slow streets with a resulting large route. It's annoying. That cycle ways would be the last option if the fast road is forbidden, or the walkaround is very large (imagine a really big park) and exists a cyclepath through the park that would be acceptable in that case.

Normally, the road has no speed limits for cyclists more than the self effort, a cyclist can ride easily at 20-30 km/h on plain terrain, 40-50 km/h (or more) on a descending, and 10-15 km/h on a climbing. Despite this, on cycle lanes, cycle paths, etc., some of them shared or at pedestrian level, with no security margins, insufficient width, maybe 2-way... the country/state/city law or your own brain said that you have to ride slow, maybe 20 km/h, maybe 10 km/h... maybe 5 if you ride with pedestrians. That's not practical at all.

Maybe a petition like mine was heard in the past. I made a search of a feature like this on other issues, but I couldn't find any similar. Furthermore, I found petitions of reinforcement the cycle paths, etc., that is the opposite that I want.

### Describe the solution you'd like

Maybe it would be great to parameterize on UI anything to do that (which kind of roads are acceptable, etc), but I think it would be enough to have a different default profile that ensures that the result route will be the best in time and length, as if I drove a car or a motorcycle, that routing would never mess around you.

The web tool [gpx.studio](https://gpx.studio) has a interesting routing profile: "racing bike", that works pretty well for my purpose (not perfect, but..). It avoids unpaved ways and also dedicated cycleways if there is an alternative better road... ok, but doesn't want to route over trunk highways (conventional roads -not motorways- or main avenues on big cities).

Motorways could be an option depending on the country. In Spain, where I live, a majority of them can be cyclable by law, if it doesn't exist an alternative conventional road for a destination.

Thanks for your attention. Regards.

**Comment by vshcherb** [comment]
Yes, there are more than 3-4 different bicycle routing profiles where you would prioritize different things for example City, MTB, Racing ...

**Comment by vodie** [comment]
You can try BRouter with the Liegerad (recumbent) profile wich avoids cycle paths. For motorways it is possible to adjust the profile.
https://brouter.de/brouter-web/#map=5/44.214/-2.021/osm-mapnik-german_style&profile=vm-forum-liegerad-schnell

BRouter can be used offline in OsmAnd via the service interface on Android, but is not very user-friendly to set up.
